### PR TITLE
Create defined crondotdee

### PR DESCRIPTION
**Pull request by surminus**
This creates a way to add stuff to `/etc/cron.d` which can be a preference from `/etc/cron.daily` and a shared crontab. In this example we'll add rkhunter as it tends to alert throughout the day when the cron.daily job doesn't run for whatever reason.

### CODE REVIEW

**Comment by rjw1** [comment]
:+1: although I do wonder if we should use an existing [module](https://forge.puppet.com/torrancew/cron), although that may be out of scope of this PR.

**Comment by surminus** [comment]
I did think about that but thought it seemed easier to write my own than install another module. Would the module you linked to clash with our own module? Would that need to be renamed or refactored?

File: modules/rkhunter/manifests/monitoring.pp [hunk 1]
```diff
@@ -44,11 +44,19 @@
...
+  file { '/usr/local/bin/rkhunter-passive-check':
     ensure  => present,
     mode    => '0755',
     content => template('rkhunter/rkhunter-passive-check.erb'),
   }
+
+  # Run it twice a day to stop superfluous alerting
+  cron::crondotdee { 'rkhunter-passive-check':
+    command => '/usr/local/bin/rkhunter-passive-check',
+    hour    => '6, 14',
```
Comment (deanwilson) [inline modules/rkhunter/manifests/monitoring.pp]: Does a space between the two hour specifiers do what you'd expect? I didn't think you could have whitespace.

File: modules/rkhunter/manifests/monitoring.pp [hunk 2]
```diff
@@ -44,11 +44,19 @@
...
+    mailto  => '',
```
Comment (deanwilson) [inline modules/rkhunter/manifests/monitoring.pp]: Should the default `mailto` be undef or an empty string? It seems odd to use both when i _think_ you just want an empty `MAILTO=`.

File: modules/cron/manifests/crondotdee.pp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# == Define: Cron::Crondotdee
+#
+# Create a file in /etc/cron.d rather than in cron.daily or a shared crontab.
+# Default behaviour is to set a command, minute and hour and to run as root.
+#
+# Note: set hour/minute/day/month/weekday as a string rather than an integer as
+# they could potentially contain non-integer values, so keep it consisent.
```
Comment (deanwilson) [inline modules/cron/manifests/crondotdee.pp]: Consistent.

**Comment by deanwilson** [comment]
Do we want the ability to specify the file mode for cronjobs created via this? `0644` by default but with the option of `0640` for things with creds in them (which I _think_ we have).

**Comment by deanwilson** [comment]
If we decide to go with our own code this looks fine. A spec or two wouldn't hurt though.

**Comment by deanwilson** [comment]
I've made a few tiny changes to this, and we've agreed to use our own module, so please re-review with the understanding we will be using this code. @rjw1 @alexmuller

File: modules/cron/manifests/crondotdee.pp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#   Which user to run as. This can be used if you wish to run a script without
+#   invoking "sudo -u $user". Default is root, as per the standard cron job.
+#
+# [*mailto*]
+#   Specifically set where the cron should attempt to email output to.
+#   Example would be a user or an email address, or use an empty string to
+#   suppress mailing any output.
+define cron::crondotdee (
+  $command,
+  $hour = '9',
```
Comment (rjw1) [inline modules/cron/manifests/crondotdee.pp]: I do wonder if these should be required parameters

File: modules/cron/templates/etc/cron.d/crondotdee.erb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<% if @mailto %>
```
Comment (rjw1) [inline modules/cron/templates/etc/cron.d/crondotdee.erb]: does this need `-`s to not generate whitespace?

**Comment by rjw1** [comment]
couple of minor questions which to be honest shouldn't stop this from being merged.

File: modules/cron/templates/etc/cron.d/crondotdee.erb (hunk 5 above)
Comment (deanwilson) [inline modules/cron/templates/etc/cron.d/crondotdee.erb]: "We want whitespace to add a visual delimiter between the jobs" he bluffs.

**Comment by rjw1** [comment]
merge when ready :+1:

**Pull request merged**